app/services/EventReportService.scala

```diff
@@ -42,24 +43,31 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
                                    overviewCacheRepository: OverviewCacheRepository
                                   ) {
 
-  private val createCompiledEventSummaryReportSchemaPath = "/resources.schemas/api-1826-create-compiled-event-summary-report-request-schema-v1.0.0.json"
-  private val compileEventOneReportSchemaPath = "/resources.schemas/api-1827-create-compiled-event-1-report-request-schema-v1.0.1.json"
-  private val compileMemberEventReportSchemaPath = "/resources.schemas/api-1830-create-compiled-member-event-report-request-schema-v1.0.4.json"
+  private val schemaPath1826 = "/resources.schemas/api-1826-create-compiled-event-summary-report-request-schema-v1.0.0.json"
+  private val schemaPath1827 = "/resources.schemas/api-1827-create-compiled-event-1-report-request-schema-v1.0.1.json"
+  private val schemaPath1830 = "/resources.schemas/api-1830-create-compiled-member-event-report-request-schema-v1.0.4.json"
 
   def compileEventReport(pstr: String, eventType: EventType)
                         (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
 
     val apiType = EventType.postApiTypeByEventType(eventType)
     val performCompile: (String, JsValue) => Future[Result] =
       apiType match {
-        case Api1826 => compileEventReportSummary _
-        case Api1827 => compileEventOneReport _
-        case Api1830 => compileMemberEventReport _
-        case _ => (_, _) => Future.successful(NoContent)
+        case Api1826 => compile1826 _
+        case Api1827 => compile1827 _
+        case Api1830 => compile1830 _
+        case api => (_, _) =>
+          Future.successful(NotImplemented(s"Compile unimplemented for API type $api (event type $eventType)"))
       }
 
     eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> apiType.toString)).flatMap {
-      case Some(data) => performCompile(pstr, data).map(_ => NoContent)
+      case Some(data) =>
+        performCompile(pstr, data).map { result =>
+          result.header.status match {
+            case NOT_IMPLEMENTED => result
```

Comment:
I think NOT_IMPLEMENTED is not the best response code for this case. 5xx are server errors but this not a server error. Request data is wrong. So 400 or maybe 405 could be better for this case.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501#:~:text=501%20is%20the%20appropriate%20response,supporting%20it%20for%20any%20resource.

Reply:
Good point. I have changed to NOT_FOUND